packages/record/src/recorder.ts

```diff
@@ -1,22 +1,39 @@
+import Data from "@lookit/data";
+import lookitS3 from "@lookit/data/dist/lookitS3";
 import autoBind from "auto-bind";
 import { JsPsych } from "jspsych";
 import { RecorderInitializeError } from "./error";
+import { getFilename } from "./utils";
 
 /** Recorder handles the state of recording and data storage. */
 export default class Recorder {
   private blobs: Blob[] = [];
+  private localDownload: boolean = false;
+  private s3: lookitS3;
+  private filename: string;
 
   /**
    * Recorder for online experiments.
    *
    * @param jsPsych - Object supplied by jsPsych.
+   * @param filename - Name for the video recording file (string). Should be
+   *   provided if the Recorder is being instantiated to start a new recording.
+   *   Can be omitted if the Recorder is being instantiated to access an
+   *   existing recording stream.
    */
-  public constructor(private jsPsych: JsPsych) {
+  public constructor(
+    private jsPsych: JsPsych,
+    filename?: string,
+  ) {
+    this.filename = filename || getFilename("null");
+    if (!this.localDownload) {
```

Comment:
I think `getFilename` can be a class method or a getter.

Reply:
Yes! I tried improving this here: cd6e347511603439a33521b255b64cd4d2eba774 The Recorder class now has a private `createFilename` method and a public getter for the filename. And now the Recorder constructor has to take a filename prefix argument.
Is that what you had in mind? Let me know if you have suggestions for improvement.